src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -0,0 +1,851 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. 
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DetectPreviewFeatureAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public partial class DetectPreviewFeatureUnitTests
+    {
+        private static async Task TestCS(string csInput)
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestState =
+                {
+                    Sources =
+                    {
+                        csInput, requiresPreviewFeaturesAttribute
+                    }
+                },
+                MarkupOptions = MarkupOptions.UseFirstDescriptor,
+            }.RunAsync();
+        }
+
+        private const string requiresPreviewFeaturesAttribute = $@"
+namespace System.Runtime.Versioning
+{{
+    [AttributeUsage(AttributeTargets.Assembly |
+                AttributeTargets.Module |
+                AttributeTargets.Class |
+                AttributeTargets.Interface |
+                AttributeTargets.Delegate |
+                AttributeTargets.Struct |
+                AttributeTargets.Enum |
+                AttributeTargets.Constructor |
+                AttributeTargets.Method |
+                AttributeTargets.Property |
+                AttributeTargets.Field |
+                AttributeTargets.Event, Inherited = false)]
+    public sealed class RequiresPreviewFeaturesAttribute : Attribute
+    {{
+        public RequiresPreviewFeaturesAttribute() {{ }}
+    }}
+}}
+";
+
+        [Fact]
+        public async Task TestPreviewMethodUnaryOperator()
+        {
+            var csInput = @" 
+using System.Runtime.Versioning; using System;
+namespace Preview_Feature_Scratch
+{" +
+@"
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            var a = new Fraction();
+            var b = [|+a|];
```

Comment:
I'm not sure why this is split into two strings being concatenated?

Reply:
There used to be another string after the `+`, but it's since been removed, so the `+` is now unnecessary. It's just painful to remove the `+` for every single test case right now, so I just left it in (since it's just unit tests)